File: Section12-15 Algorithms/Section13 Sorting/166 Selection Sort.js
const numbers = [99, 44, 6, 2, 1, 5, 63, 87, 283, 4, 0];

//For each iteration, find the smallest of the remaining array, swap it with the first element of the remaining array before doing next iteration
function selectionSort(array) {
  const length = array.length;
  for(let i=0; i<length; i++){
    let minIndex = i;
    let temp = array[i];

    for(let j=i+1; j<length; j++){
      if(array[j] < array[minIndex]){
        minIndex = j;
      }
    }

    array[i] = array[minIndex];
    array[minIndex] = temp;
  }
  return array;
}

selectionSort(numbers);

//[ 0, 1, 2, 4, 5, 6, 44, 63, 87, 99, 283 ]
console.log(numbers);

File: Section12-15 Algorithms/Section13 Sorting/166 Selection Sort.java
class Main {
  public static void main(String[] args) {
    int[] numbers = {99, 44, 6, 2, 1, 5, 63, 87, 283, 4, 0};

    int length = numbers.length;

		for(int i=0; i<length; i++) {
			int minIndex = i;
			int temp = numbers[i];

			for(int j=i+1; j<length; j++) {
				if(numbers[j] < numbers[minIndex])
					minIndex =j;
			}

			numbers[i] = numbers[minIndex];
			numbers[minIndex] = temp;
		}

		for(int i: numbers) {
			System.out.print(" " + i);
		}
	}
}
